File: loquacious/src/main/java/com/saantiaguilera/loquacious/model/Quantity.java
package com.saantiaguilera.loquacious.model;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.ibm.icu.text.PluralRules;

/**
 * Created by saguilera on 11/18/17.
 */

public enum Quantity {
    ZERO,
    ONE,
    TWO,
    FEW,
    MANY,
    OTHER;

    @Override
    public String toString() {
        return super.toString().toLowerCase();
    }

    @Nullable
    public static Quantity from(@NonNull PluralRules rules, int quantity) {
        String plural = rules.select(quantity).toLowerCase();
        for (Quantity enumerable : values()) {
            if (enumerable.toString().contentEquals(plural)) {
                return enumerable;
            }
        }
        return null;
    }
}

File: loquacious/src/main/java/com/saantiaguilera/loquacious/model/Quantity.kt
package com.saantiaguilera.loquacious.model

import android.support.annotation.RestrictTo
import com.ibm.icu.text.PluralRules

/**
 * Created by saguilera on 11/18/17.
 */
enum class Quantity {
    ZERO,
    ONE,
    TWO,
    FEW,
    MANY,
    OTHER;

    override fun toString(): String = super.toString().toLowerCase()

    @RestrictTo(RestrictTo.Scope.LIBRARY)
    companion object {

        fun from(rules: PluralRules, quantity: Int): Quantity? {
            val plural by lazy { rules.select(quantity.toDouble()).toLowerCase() }
            return values().firstOrNull { it.toString().contentEquals(plural) }
        }
    }
}
